scripts/koji/koji_build.py

```diff
@@ -53,13 +53,17 @@ def check_commit_is_available_remotely(dirpath, hash, target, warn):
         if target is not None and re.match(r'v\d+\.\d+-u-.+', target):
             raise Exception("Building with a user target requires using --pre-build or --test-build.\n")
         try:
-            expected_branch = PROTECTED_TARGETS.get(target)
-            if (
-                expected_branch is not None
-                and not is_remote_branch_commit(dirpath, hash, expected_branch)
-            ):
-                raise Exception(f"The current commit is not the last commit in the remote branch {expected_branch}.\n"
-                                f"This is required when using the protected target {target}.\n")
+            available = False
+            supported_branches = PROTECTED_TARGETS.get(target)
+            if supported_branches:
```

Comment:
`if supported_branches is not None` would leave the ability of forbidding any build on a branch by setting its branch list to `[]`, otherwise just just removing the last branch from the list would have a likely unexpected result.

Reply:
> No problem to add it, it will work but I think is not necessary, or am I missing something ?

With `if supported_branches:` both `None` and `[]` are handled the same, that is an explicit empty list of branches will be synonymous to a target not being listed as protected.  The latter case denotes a non-protected target, so this `if` causes the check to be (correctly) skipped.  However, the semantics of having a target associated with a list of branch is that when we add/remove to the list it respectively causes more/less branches to be allowed.  And if `[]` is synonymous to `None`, removing the last branch, instead of removing that branch from the allowed ones, causes all branches to become valid (the "unexpected result" I was referring to).